From 2784aa3683ae70d670bbde74b023a639d9cbd0da Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Fri, 6 May 2022 10:41:52 +0530
Subject: [PATCH] libstagefright: omx: Add support for loading prebuilt ddp and
 ac4 decoder lib

[HELLBOY017]: Also allow to load dolby codec in MediaCodecList

Change-Id: I3b85c15889c9e3aef152e726a83936234103254e
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 media/libstagefright/MediaCodecList.cpp    | 1 +
 media/libstagefright/omx/SoftOMXPlugin.cpp | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/media/libstagefright/MediaCodecList.cpp b/media/libstagefright/MediaCodecList.cpp
index 4ad3276743..03bf12e6dc 100644
--- a/media/libstagefright/MediaCodecList.cpp
+++ b/media/libstagefright/MediaCodecList.cpp
@@ -319,6 +319,7 @@ const sp<AMessage> MediaCodecList::getGlobalSettings() const {
 //static
 bool MediaCodecList::isSoftwareCodec(const AString &componentName) {
     return componentName.startsWithIgnoreCase("OMX.google.")
+            || componentName.startsWithIgnoreCase("OMX.dolby.")
             || componentName.startsWithIgnoreCase("c2.android.")
             || (!componentName.startsWithIgnoreCase("OMX.")
                     && !componentName.startsWithIgnoreCase("c2."));
diff --git a/media/libstagefright/omx/SoftOMXPlugin.cpp b/media/libstagefright/omx/SoftOMXPlugin.cpp
index 8c186c90f8..25b36cb7b6 100644
--- a/media/libstagefright/omx/SoftOMXPlugin.cpp
+++ b/media/libstagefright/omx/SoftOMXPlugin.cpp
@@ -61,6 +61,10 @@ static const struct {
     { "OMX.google.flac.decoder", "flacdec", "audio_decoder.flac" },
     { "OMX.google.flac.encoder", "flacenc", "audio_encoder.flac" },
     { "OMX.google.gsm.decoder", "gsmdec", "audio_decoder.gsm" },
+    { "OMX.dolby.ac3.decoder", "ddpdec", "audio_decoder.ac3" },
+    { "OMX.dolby.eac3-joc.decoder", "ddpdec", "audio_decoder.eac3_joc" },
+    { "OMX.dolby.eac3.decoder", "ddpdec", "audio_decoder.eac3" },
+    { "OMX.dolby.ac4.decoder", "ac4dec", "audio_decoder.ac4" },
 };
 
 static const size_t kNumComponents =
